src/pip/_internal/resolution/resolvelib/factory.py

```diff
@@ -133,8 +133,12 @@ def _fail_if_link_is_unsupported_wheel(self, link: Link) -> None:
         wheel = Wheel(link.filename)
         if wheel.supported(self._finder.target_python.get_tags()):
             return
-        msg = f"{link.filename} is not a supported wheel on this platform."
-        raise UnsupportedWheel(msg)
+        msg = (f"{link.filename} (tags {wheel.file_tags}) is not a supported "
+                "wheel on the platform "
+               f"{self._finder.target_python.get_tags()}.")
+        logger.debug(msg)
+        raise UnsupportedWheel(f"{link.filename} (tags {wheel.file_tags}) is "
+                                "not a supported wheel on this platform.")
```

Comment:
This is not the right place for this. `UnsupportedWheel` is caught in many places in the code base, and we don’t want this message in those places. The message should only be emitted when `UnsupportedWheel` is *uncaught*, which should be done here:

https://github.com/pypa/pip/blob/4704da4dad58ca89439edc109f7d12217c0fd447/src/pip/_internal/cli/base_command.py#L172-L214

Since this frame is lacking context for the message, we need to create a new exception class like this:

```python
class UnsupportedWheelTag(UnsupportedWheel):
    def __init__(self, wheel, supported_tags):
        super().__init__()
        self.wheel = wheel
        self.supported_tags = supported_tags

    def __str__(self):
        return (
            f"{self.wheel.filename} (tags {self.wheel.file_tags}) is "
            f"not a supported wheel on this platform."
        )
```

And add an `except` block like this:

```python
except UnsupportedWheelTag as e:
    logger.debug("Compatible tags for this platform: f{e.supported_tags}")
    raise
```

I trimmed the message a little since the exception’s error message (which would be immediately below this anyway) already contains wheel filename and tags.

Does this sound reasonable?